tests/unit/hbar_test.py

```
@@ -200,4 +218,13 @@ def test_factory_methods():
     assert Hbar.from_gigabars(-1).to_hbars() == -1_000_000_000
-    assert Hbar.from_gigabars(1).to_tinybars() == 100_000_000_000_000_000
\ No newline at end of file
+    assert Hbar.from_gigabars(1).to_tinybars() == 100_000_000_000_000_000
+
+@pytest.mark.parametrize(
+    'invalid_tinybars',
+    ['1', 0.1, Decimal('0.1'), True, False, object, {}]
+)
+def test_from_tinybars_invalid_type_param(invalid_tinybars):
+    """Test from_tinybar method raises error if the type is not int."""
+    with pytest.raises(TypeError, match="tinybars must be an int."):
```

Ruff (0.14.11)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)